### PR TITLE
fenwick_tree.java: Add Fenwick Tree Data Structure

### LINKED ISSUE #218: [Data Structure] Fenwick Tree [Java]
The FenwickTree is a data structure for O(log(n)) updates and queries that answers both point-update and range-query questions, find out more here: https://en.wikipedia.org/wiki/Fenwick_tree
It would be nice to have this data structure implemented in Java (that is used a lot in programming contexts)

### PR DESCRIPTION
**Pull request by SnowyCoder**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

[Closes #218](https://github.com/NITSkmOS/Algorithms/issues/218)

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [x] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [x] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [x] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [x] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [x] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

**Comment by SnowyCoder** [comment]
The `&nbsp;` code is a workaround for [this issue](https://github.com/atom/markdown-preview/issues/206)

**Comment by SnowyCoder** [comment]
Hello, are there any updates on this issue? I think I followed all of the guidelines

File: fenwick_tree/python/fenwicktree.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+
```
Comment (sangamcse) [inline fenwick_tree/python/fenwicktree.py]: One Issue per PR.

**Comment by SnowyCoder** [comment]
Ok I misread the last request and I thought that I needed to merge the two PRs.
I'll fix once I get home

**Comment by SnowyCoder** [comment]
Ok I misread the last request and I thought that I needed to merge the two PRs.
I'll fix once I get home

**Pull request merged**